bundle/src/main/java/com/adobe/acs/commons/wcm/impl/SiteMapServlet.java

```diff
@@ -139,6 +160,39 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
         }
     }
 
+    private Collection<Resource> getAssetFolders(Page page, ResourceResolver resolver) {
+        Map<String, Resource> allAssetFolders = new TreeMap<String, Resource>();
+        ValueMap properties = page.getProperties();
+        String[] configuredAssetFolderPaths = properties.get(damAssetProperty, String[].class);
+        if (configuredAssetFolderPaths != null) {
+            for (String configuredAssetFolderPath : configuredAssetFolderPaths) {
+                if (StringUtils.isNotBlank(configuredAssetFolderPath)) {
+                    Resource configuredAssetFolder = resolver.getResource(configuredAssetFolderPath);
+                    if (configuredAssetFolder != null && configuredAssetFolder.isResourceType("sling:OrderedFolder")) {
```

Comment:
I assume because  I ootb DAM folders are ordered- but agree. IMO the Folders aren't what's interesting; it should be walking the tree for dam:Assets..  The collection method is double duty: first collecting all sub folders and then iterate over this folder list for assets. .. This is what using the resource visitor comment (above) is about.